tests/lib/pkgdb.sh

```diff
@@ -305,19 +305,125 @@ distro_install_build_snapd(){
-export DISTRO_BUILD_DEPS=()
-case "$SPREAD_SYSTEM" in
-    debian-*|ubuntu-*)
-        DISTRO_BUILD_DEPS=(build-essential curl devscripts expect gdebi-core jq rng-tools git netcat-openbsd)
-        ;;
-    fedora-*)
-        DISTRO_BUILD_DEPS=(mock git expect curl golang rpm-build redhat-lsb-core)
-        ;;
-    opensuse-*)
-        DISTRO_BUILD_DEPS=(osc git expect curl golang-packaging lsb-release netcat-openbsd jq rng-tools)
-        ;;
-    *)
-        ;;
-esac
+
+pkg_dependency_ubuntu_generic(){
+    echo autoconf
+    echo automake
+    echo autotools-dev
+    echo build-essential
+    echo curl
+    echo devscripts
+    echo expect
+    echo gdebi-core
+    echo git
+    echo indent
+    echo jq
+    echo libapparmor-dev
+    echo libglib2.0-dev
+    echo libseccomp-dev
+    echo libudev-dev
+    echo netcat-openbsd
+    echo pkg-config
+    echo python3-docutils
+    echo rng-tools
+    echo udev
+}
+
+pkg_dependency_ubuntu_classic(){
+    echo cups
+    echo dbus-x11
+    echo jq
+    echo man
+    echo printer-driver-cups-pdf
+    echo python3-yaml
+    echo upower
+
+    case "$SPREAD_SYSTEM" in
+        ubuntu-14.04-*)
+            echo "linux-image-extra-$(uname -r)"
+            echo pollinate
+            ;;
+        ubuntu-16.04-32)
+            echo "linux-image-extra-$(uname -r)"
+            echo pollinate
+            ;;
+        ubuntu-16.04-64)
+            echo gccgo-6
+            echo kpartx
+            echo libvirt-bin
+            echo "linux-image-extra-$(uname -r)"
+            echo pollinate
+            echo qemu
+            echo x11-utils
+            echo xvfb
+            ;;
+        ubuntu-*)
+            echo "linux-image-extra-$(uname -r)"
+            echo pollinate
+            ;;
+        debian-*)
+            ;;
+    esac
+}
+
+pkg_dependency_ubuntu_core(){
+    echo "linux-image-extra-$(uname -r)"
+    echo pollinate
+}
+
+pkg_dependency_fedora(){
+    echo curl
+    echo expect
+    echo git
+    echo golang
+    echo mock
+    echo redhat-lsb-core
+    echo rpm-build
+}
+
+pkg_dependency_opensuse(){
+    echo curl
+    echo expect
+    echo git
+    echo golang-packaging
+    echo jq
+    echo lsb-release
+    echo netcat-openbsd
+    echo osc
+    echo rng-tools
+}
+
+pkg_dependencies(){
+    case "$SPREAD_SYSTEM" in
+        ubuntu-core-16-*)
+            pkg_dependency_ubuntu_generic
+            pkg_dependency_ubuntu_core
+            ;;
+        ubuntu-*|debian-*)
+            pkg_dependency_ubuntu_generic
+            pkg_dependency_ubuntu_classic
+            ;;
+        fedora-*)
+            pkg_dependency_fedora
+            ;;
+        opensuse-*)
+            pkg_dependency_opensuse
+            ;;
+        *)
+            ;;
+    esac  
+}
+
+install_dependencies(){
+    pkgs=$(pkg_dependencies | tr "\n" " ")
+
+    # ensure systemd is up-to-date, if there is a mismatch libudev-dev
+    # will fail to install because the poor apt resolver does not get it
```

Comment:
As such, this should be inside distro_install_package and should be conditional on libudev-dev being part of the requested set.